drivers/imx-enet.c

```diff
@@ -157,25 +174,57 @@ static void enet_start(enet_state_t *state)
 #endif
 #if ENET_PROMISC_MODE
 		ENET_RCR_PROM |
+#endif
+#if ENET_DIS_RX_ON_TX
+		ENET_RCR_DRT |
+#endif
+#if ENET_MAC_CONTROL_FRAME_ENABLE
+		ENET_RCR_CFEN |
+#endif
+#if ENET_INTERNAL_LOOPBACK_MODE && !ENET_RMII_MODE && !ENET_DIS_RX_ON_TX
+		ENET_RCR_LOOP |
 #endif
 		ENET_RCR_MII_MODE;
-	state->mmio->RACC = ENET_RACC_SHIFT16 |
+	/* RCR */
+
+	state->mmio->RACC =
+#if ETH_PAD_SIZE == 2
+		ENET_RACC_SHIFT16 |
+#endif
 #if !ENET_PROMISC_MODE
 		ENET_RACC_LINEDIS | ENET_RACC_PRODIS | ENET_RACC_IPDIS |
 #endif
 		ENET_RACC_PADREM;
+	/* RACC */
 
-	state->mmio->TCR = ENET_TCR_FDEN;
+	state->mmio->TACC =
+#if ENET_ENABLE_CHECKSUM
+		ENET_TACC_PROCHK |
+#endif
+#if ENET_ENABLE_IPCHK
+		ENET_TACC_IPCHK |
+#endif
 #if ETH_PAD_SIZE == 2
-	state->mmio->TACC = ENET_TACC_SHIFT16;
-#else
-	state->mmio->TACC = 0;
+		ENET_TACC_SHIFT16 |
+#endif
+		0x0;
+	/* TACC */
+
+	state->mmio->TCR =
+#if ENET_SET_MAC_ON_TX
+		ENET_TCR_ADDINS |
```

Comment:
Yes, it works as expected - sets the MAC address in the frame on TX. Should it maybe be a part of another commit or is it not needed at all in your opinion?

Reply:
I just want to avoid adding untested code paths (especially as a part of the commit which should add only a port for `rt106x` as it's disabled for this target).

Did you enable this feature, comment out the parts in lwip stack setting the ETH hw address (or purposely write incorrect data here) - probably in `ethernet_output` and verify on other device (eg. using Wireshark) that the resulting packets have the correct ETH address? Please confirm (the same applies for all other extra features you've added).

Regarding if it's needed - currently there is no graceful way to use it in `lwip` (as per testing path - you would need to manually change the `lwip` code - you should already know that if you have tested it) - I don't expect this feature to be enabled/used soon. It might be useful if/when we would port some other TCP/IP stack so I'm not against adding it if it works.

Reply:
ok, it will be added later if needed then